compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs

```diff
@@ -491,6 +509,29 @@ timedSection targetDiffTime block = do
         throwError $ TimedSectionTookTooLong targetDiffTime actualDiffTime
     return value
 
+
+templateChoicesToProps :: TemplateChoices -> TemplateProp
+templateChoicesToProps tca = TemplateProp choicesInTpl (sum $ map (length . actions ) (choiceAndActions tca))
+    where choicesInTpl = Set.fromList $ map (\ca -> ExpectedChoices ( DAP.renderPretty $ choiceName ca) (choiceConsuming ca)) (choiceAndActions tca)
+
+graphTest :: LF.World -> LF.Package -> Set.Set TemplateProp -> Either [TemplateProp] ()
+graphTest wrld lfPkg expectedProps =
+    if expectedProps == (Set.fromList $ map templateChoicesToProps tplPropsActual)
+        then Right ()
+        else Left $ map templateChoicesToProps tplPropsActual
+    where tplPropsActual = concatMap (moduleAndTemplates wrld) (NM.toList $ LF.packageModules lfPkg)
+
+expectedPoperties :: D.NormalizedFilePath -> Set.Set TemplateProp -> ShakeTest ()
+expectedPoperties damlFilePath expectedProps = do
+    ideState <- ShakeTest $ Reader.asks steService
+    mbDalf <- liftIO $ API.runAction ideState (API.getDalf damlFilePath)
```

Comment:
I would add an `expectNoErrors` call here. That way you get usable diagnostics if it fails.
And then, I would just turn the `Nothing` case into a pattern match failure, e.g.,
`Just dalf <- pure mbDalf`

Reply:
Thanks for the help with this, Using monad fail now.